### PR TITLE
Implementing an Epsilon in Digital Filtering

### PR DESCRIPTION
**Pull request by anistouri**
Signed-off-by: TOURI ANIS <anis-1.touri@rte-france.com>

### CODE REVIEW

File: src/hooks/use-aggrid-local-row-filter.ts [hunk 1]
```diff
@@ -90,14 +118,74 @@ export const useAggridLocalRowFilter = (
...
+                // Attempt to convert filter value to a number if it's a string, otherwise keep it as is
+                let valueAsNumber: number | string[] =
+                    typeof filter.value === 'string'
+                        ? parseFloat(filter.value)
+                        : filter.value;
+                // If the value is successfully converted to a number, apply tolerance adjustments
+                if (typeof valueAsNumber === 'number') {
+                    let facteur = Math.pow(10, 5);
+                    // Truncate the number to maintain precision
+                    let nombre1Tronque =
```
Comment (klesaulnier) [inline src/hooks/use-aggrid-local-row-filter.ts:146]: try to use variables in english
factor
truncatedNumber

File: src/hooks/use-aggrid-local-row-filter.ts [hunk 2]
```diff
@@ -90,14 +118,74 @@ export const useAggridLocalRowFilter = (
...
+                    // Depending on the filter type, adjust the filter value by adding or subtracting the tolerance
+                    switch (filter.type) {
+                        case 'notEqual':
+                            // For 'notEqual', create two conditions: one for greaterThan and one for lessThan
+                            return [
+                                {
+                                    ...filter,
+                                    type: 'greaterThan',
+                                    value: (nombre1Tronque + tolerance).toFixed(
+                                        5
```
Comment (klesaulnier) [inline src/hooks/use-aggrid-local-row-filter.ts:157]: I think this 5 is also linked to tolerance, or at least to "facteur"
make them linked with a CONSTANT

File: src/hooks/use-aggrid-local-row-filter.ts [hunk 3]
```diff
@@ -90,14 +118,74 @@ export const useAggridLocalRowFilter = (
...
+                                    ),
+                                },
+                                {
+                                    ...filter,
+                                    type: 'lessThan',
+                                    value: (nombre1Tronque - tolerance).toFixed(
+                                        5
```
Comment (klesaulnier) [inline src/hooks/use-aggrid-local-row-filter.ts:164]: same here

File: src/hooks/use-aggrid-local-row-filter.ts [hunk 4]
```diff
@@ -90,14 +118,74 @@ export const useAggridLocalRowFilter = (
...
+                                    ),
+                                },
+                            ];
+                        case 'lessThanOrEqual':
+                            // For 'lessThanOrEqual', adjust the value upwards by the tolerance
+                            return {
+                                ...filter,
+                                value: (nombre1Tronque + tolerance).toFixed(5),
```
Comment (klesaulnier) [inline src/hooks/use-aggrid-local-row-filter.ts:172]: same here

File: src/hooks/use-aggrid-local-row-filter.ts [hunk 5]
```diff
@@ -90,14 +118,74 @@ export const useAggridLocalRowFilter = (
...
+                            };
+                        case 'greaterThanOrEqual':
+                            // For 'greaterThanOrEqual', adjust the value downwards by the tolerance
+                            return {
+                                ...filter,
+                                value: (nombre1Tronque - tolerance).toFixed(5),
```
Comment (klesaulnier) [inline src/hooks/use-aggrid-local-row-filter.ts:178]: same here

File: src/hooks/use-aggrid-local-row-filter.ts [hunk 6]
```diff
@@ -90,14 +118,74 @@ export const useAggridLocalRowFilter = (
...
+                let valueAsNumber: number | string[] =
+                    typeof filter.value === 'string'
+                        ? parseFloat(filter.value)
+                        : filter.value;
+                // If the value is successfully converted to a number, apply tolerance adjustments
+                if (typeof valueAsNumber === 'number') {
+                    let facteur = Math.pow(10, 5);
+                    // Truncate the number to maintain precision
+                    let nombre1Tronque =
+                        Math.floor(valueAsNumber * facteur) / facteur;
```
[the remark this answers is not in the document]
Comment (anistouri) [reply src/hooks/use-aggrid-local-row-filter.ts]: The purpose of the code is to reduce the number of decimal places of a given number without rounding

File: src/hooks/use-aggrid-local-row-filter.ts [hunk 7]
```diff
@@ -43,24 +43,52 @@ export const useAggridLocalRowFilter = (
...
+                        };
+
+                        // Set up the filter model with an 'OR' operator, including both existing and new conditions
+                        agGridFilterModel[filter.column] = {
+                            filterType: 'number',
+                            type: 'number',
+                            operator: 'OR',
+                            condition1: existingCondition, // Use the existing condition
+                            condition2: newCondition, // Add the new condition
+                        };
```
[the remark this answers is not in the document]
Comment (anistouri) [reply src/hooks/use-aggrid-local-row-filter.ts:78]: It's to transform the notEquals into two conditions, greater and lesser

File: src/hooks/use-aggrid-local-row-filter.ts [hunk 8]
```diff
@@ -43,24 +43,52 @@ export const useAggridLocalRowFilter = (
...
+                    } else {
+                        // If no existing filter for this column, create a new filter configuration
+                        agGridFilterModel[filter.column] = {
+                            filterType: filter.dataType,
+                            type: filter.type,
+                            filter:
+                                filter.dataType === FILTER_DATA_TYPES.NUMBER
```
[the remark this answers is not in the document]
Comment (anistouri) [reply src/hooks/use-aggrid-local-row-filter.ts:85]: We can be in the else case when agGridFilterModel[filter.column] is false

File: src/hooks/use-aggrid-local-row-filter.ts [hunk 9]
```diff
@@ -90,14 +118,74 @@ export const useAggridLocalRowFilter = (
...
+        return filters
+            .map((filter): FilterSelectorType | FilterSelectorType[] => {
+                // Attempt to convert filter value to a number if it's a string, otherwise keep it as is
+                let valueAsNumber: number | string[] =
+                    typeof filter.value === 'string'
+                        ? parseFloat(filter.value)
+                        : filter.value;
+                // If the value is successfully converted to a number, apply tolerance adjustments
+                if (typeof valueAsNumber === 'number') {
+                    let facteur = Math.pow(10, 5);
```
[the remark this answers is not in the document]
Comment (anistouri) [reply src/hooks/use-aggrid-local-row-filter.ts]: done

File: src/hooks/use-aggrid-local-row-filter.ts (hunk 1 above)
Comment (anistouri) [reply src/hooks/use-aggrid-local-row-filter.ts]: My bad, I'll fix it

File: src/hooks/use-aggrid-local-row-filter.ts (hunk 3 above)
Comment (anistouri) [reply src/hooks/use-aggrid-local-row-filter.ts]: done

File: src/hooks/use-aggrid-local-row-filter.ts (hunk 4 above)
Comment (anistouri) [reply src/hooks/use-aggrid-local-row-filter.ts]: done

File: src/hooks/use-aggrid-local-row-filter.ts (hunk 5 above)
Comment (anistouri) [reply src/hooks/use-aggrid-local-row-filter.ts]: done

File: src/hooks/use-aggrid-local-row-filter.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (souissimai) [reply src/hooks/use-aggrid-local-row-filter.ts]: facteur ==> factor maybe?

**Review by souissimai — APPROVED** [review]
test and code ok

File: src/hooks/use-aggrid-local-row-filter.ts (hunk 2 above)
Comment (anistouri) [reply src/hooks/use-aggrid-local-row-filter.ts]: done

**Review by klesaulnier — APPROVED** [review]

**Pull request merged**